src/rgw/rgw_rest.cc

```diff
@@ -139,20 +139,21 @@ map<int, const char *> http_status_names;
  */
 string lowercase_underscore_http_attr(const string& orig)
 {
-  const char *s = orig.c_str();
-  char buf[orig.size() + 1];
-  buf[orig.size()] = '\0';
-
-  for (size_t i = 0; i < orig.size(); ++i, ++s) {
-    switch (*s) {
-      case '-':
-        buf[i] = '_';
-        break;
-      default:
-        buf[i] = tolower(*s);
-    }
-  }
-  return string(buf);
+  std::string mod;
+  mod.reserve(orig.size());
+  std::transform(orig.cbegin(),
```

Comment:
+1 ALTHOUGH in this case since there's only one alternative a ternary or if-else might be more concise (but I'm cool with this as written, I'm a fan of map()! And by map(), I mean transform(). :-)

Reply:
Consider-also the ranges version.

Reply:
I think ranges are one of the things that tends not to work thanks to the unending reign of GCC11.